upup/pkg/fi/cloud.go

```diff
@@ -169,3 +173,41 @@ func GuessCloudForZone(zone string) (CloudProviderID, bool) {
 	c, found := zonesToCloud[zone]
 	return c, found
 }
+
+// for sorting ip addresses
+type AddrSlice []address.Address
+
+func (s AddrSlice) Len() int           { return len(s) }
+func (s AddrSlice) Less(i, j int) bool { return s[i] < s[j] }
+func (s AddrSlice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+
+// input vpc info, output biggest free subnet(not aws subnet, bu ipv4) in vpc
+func (vpcInfo *VPCInfo) FindBiggestFreeSubnet() string {
+	var arr AddrSlice
+	// add min and max for vpc
+	net, _ := address.ParseCIDR(vpcInfo.CIDR)
+	arr = append(arr, net.Start())
+	arr = append(arr, net.End()-1)
```

Comment:
why we need care about dns and other reserved ip ? we just take big vpc subnet, subtract existed subnet and use biggest free subnet for k8s. all reserved ip addresses inside subnet is problem of aws. or i something missed ?